### PR TITLE
Enable Flat beams in Beam.generateBeams()

### PR DESCRIPTION
**Pull request by aaronmars**
This change allows beam slope calculations to be bypassed in the beam generation logic, and allows for a flat beam.  The current, default behavior of generateBeams() is unchanged.

The offset for the flat beam can be universally applied for all of the notes sent to generateBeams(), or it can be automatically calculated per beam group.

To enable flat beams with different heights per beam group, send in `flat_beams: true` to the generateBeams() options.

If the same height is desired for all of the beams in the set of notes sent to generateBeams(), a 2nd parameter, `flat_beam_offset`, can be passed along with `flat_beams: true`.  This parameter is ignored if `flat_beams` is `false` (the default).

I added 5 tests for the 2 modes described above with both up-stems and down-stems (and mixed, up and down).  I tried to put in as many edge cases as I could think of.

### CODE REVIEW

File: src/beam.js [hunk 1]
```diff
@@ -71,6 +71,7 @@ Vex.Flow.Beam = (function() {
       this.notes = notes;
       this.beam_count = this.getBeamCount();
       this.break_on_indices = [];
+      this.min_flat_beam_offset = 15;
```
Comment (0xfe) [inline src/beam.js]: Can you put this in `render_options`?

**Comment by 0xfe** [comment]
Thanks for the commits. Can you paste in some screenshots of the test output here please?

In particular, I'm curious about the `flat_beam_offset` variable and how it affects the beams.

**Comment by aaronmars** [comment]
Will do.  I'll also address the inline comment above.
Basically, `flat_beam_offset` can mess up the rendering if it is forced beyond the note heads (depending on the stem direction).

The 2 options to fix this (that I see) are:

* Override the supplied offset if it is going to cause this out-of-bounds issue.
* Throw an error if this occurs.

**Comment by aaronmars** [comment]
The new functionality allows for flat_beam_offset to be ignored if it is going to cause a collision with the note heads.
Here are the tests (I added 2 since the original PR was sent).
flat_beams_tests_1
flat_beams_tests_2
flat_beams_tests_3

**Comment by 0xfe** [comment]
Thanks for the commit. Looks good.
